### ISSUE TITLE
参考文献リスト

### ISSUE THREAD

**Issue by Geson-anko** [Geson-anko/RWKV-ReImpl#6]
このissueにはRWKVの再現実装のための参考文献をコメントを記述する形で追加していきます。  
ほかの人もわかりやすくするために、参考文献の内容を簡潔に要約してください。

**Comment by Geson-anko** [comment]
## RWKVの公式情報
### ブログ
- overview: <https://johanwind.github.io/2023/03/23/rwkv_overview.html>
- details: <https://johanwind.github.io/2023/03/23/rwkv_details.html>

### 公式リポジトリ
<https://github.com/BlinkDL/RWKV-LM>

**Comment by Geson-anko** [comment]
## 日本語のLLM用ベンチマークタスク集 JGLUE
収録されているベンチマークは4つ。
- MARC-ja: 文章のnega-posi分類
- JSTS/JNLI: 2文中の類似度や関係分類
- JSQuAD: 抜き出し質問回答
- JCommonsenseQA: 常識QA
https://techblog.yahoo.co.jp/entry/2022122030379907/

**Comment by Geson-anko** [comment]
## 日本語Wiki コーパスの使用方法
https://note.com/npaka/n/n30a0cef78f42

**Comment by Geson-anko** [comment]
## Tokenizer
> ## 日本語Wiki コーパスの使用方法
> https://note.com/npaka/n/n30a0cef78f42

sentencepieceを使用し、Tokenizeします。  
https://github.com/google/sentencepiece

**Comment by Geson-anko** [comment]
## Hyper Parameters
公式のRWKVのハイパーパラメータに関して

- float mode: bfloat16
- layer数: 6
- ctx_len: 1024
- batch_size: 12
- lr_init: 8e-4 (For L12-D768, set lr_init = 6e-4. For L24-D1024, set lr_init = 4e-4. For L24-D2048, set lr_init = 3e-4.)
- lr_final: 1e-5 (指数関数的に減衰)
- steps per epoch: 10000
- warmup token数: 0 (途中から行う場合は  `50 * ctx_len * batch_size // NUM_GPUS`)
- optimizer: Adam
  - betas: (0.9, 0.99)
  - eps: 1e-8

### 参考
https://github.com/BlinkDL/RWKV-LM/blob/main/RWKV-v4/train.py

**Comment by Geson-anko** [comment]
## RWKVの重み初期化に関して

### 参考文献
- `torch.nn.init.orthogonal_`:  
  https://pytorch.org/docs/stable/nn.init.html?highlight=orthogonal#torch.nn.init.orthogonal_
  https://arxiv.org/abs/1312.6120
